eventservice/api/controllers/events.py

```diff
@@ -414,43 +392,31 @@ def patch(event_id):
 
 
 def delete(event_id):
-    can_delete = False
-
+    is_all_group_admin = False
     # check permission if the user has access to all group events.
     try:
-        is_all_group_event = check_all_group_event_admin()
-        if is_all_group_event:
-            can_delete = True
+        is_all_group_admin = check_all_group_event_admin()
     except Exception as ex:
         msg = "Failed to parse the id token."
         __logger.exception(msg, ex)
         abort(500)
 
-    if not can_delete:
-        auth_middleware.authorize(auth_middleware.ROKWIRE_EVENT_WRITE_GROUPS)
-
     if not ObjectId.is_valid(event_id):
         abort(400)
-
-    group_memberships = list()
-    try:
-        _, group_memberships = get_group_memberships()
-    except Exception as ex:
-        __logger.exception(msgs.ERR_MSG_GET_GROUP_MEMBERSHIP)
-        abort(500)
-
-    db = None
-    event = None
-    try:
-        db = get_db()
-        event = db['events'].find_one({'_id': ObjectId(event_id)}, {'_id': 0})
-    except Exception as ex:
-        __logger.exception(msgs.ERR_MSG_GET_EVENT)
-        abort(500)
-
-    if not can_delete:
-        if not check_group_event_admin_access(event, group_memberships):
-            abort(401)
+    # check additional permission if the user has not access to all group events.
+    if not is_all_group_admin:
+        try:
+            db = get_db()
+            event = db['events'].find_one({'_id': ObjectId(event_id)}, {'_id': 0})
```

Comment:
```suggestion
            event = db['events'].find_one({'_id': ObjectId(event_id)}, {'_id': 0})
            
            # Event not found
            if not event:
                abort(404)
                
```
If the event is not found after querying from the database, I think it will be better to return a 404 error. Otherwise, it will try to call `None.get('createdByGroupId'):` in line 412 and return an internal server error.

Reply:
Probably the below change is better as with the above suggestion, the endpoint still returns HTTP 500 due to the exception handling. The below lines of code will return HTTP 404 if the event is not present and still return HTTP 500 in case of other errors. Please let me know your thoughts. Thanks.

```
    event = None
    # check additional permission if the user has not access to all group events.
    if not is_all_group_admin:
        try:
            db = get_db()
            event = db['events'].find_one({'_id': ObjectId(event_id)}, {'_id': 0})
        except Exception as ex:
            __logger.exception(msgs.ERR_MSG_GET_EVENT)
            abort(500)

        # Event not found
        if not event:
            abort(404)

        # check group event
        if event.get('createdByGroupId'):
            group_event_admin_operation_permission_check(event, event_id)
        # check non-group event
        else:
            auth_middleware.authorize(auth_middleware.ROKWIRE_EVENT_WRITE_GROUPS)
```